### DISCUSSION TITLE
Publishing VM images to public

### DISCUSSION THREAD

**Discussion by encX** [actions/virtual-environments#3919] [Ideas]
Hi, I'm working projects trying to use GitHub actions at scale and wanted to have self-hosted runners fully compatible with what GitHub provides.

I know that I can build VM images out of the scripts myself but I have hard time building one and keeping tools inside the images up to date is also challenging.

So, Is it possible to have images that GitHub already built for GitHub-hosted runners shared publicly?

**Comment by maxim-lobanov** [comment]
Hello @encX , we have already considered this opportunity earlier.
Unfortunately, it is not possible for now due to the legal issues. License agreement of some software installed on images don't allow us to distribute ready images somehow.
Some additional context can be found in https://github.com/actions/virtual-environments/issues/176#issuecomment-620638942

**Reply by encX** [reply]
@maxim-lobanov Understood. Thank you for your quick response. Appreciated it!

**Reply by dicko2** [reply]
@maxim-lobanov would it be a legal issue if we forked the repo and built it ourselves in a separate azure account? seems odd that there would be a legal issue with an open source repo, it seems that all the software that is install is publicly available in blobs that are listed in the repo itself

**Reply by catthehacker** [reply]
It's not a problem to redistribute scripts for software installation but there are licences that prohibit redistribution of the software itself.

**Reply by dicko2** [reply]
Which software?

**Reply by dicko2** [reply]
And i guess a bit more to the point, is there any support to help GHE customers for building runners on prem?

**Reply by maxim-lobanov** [reply]
If you build own images for your team, there is no legal issues.
If you decided to distribute ready images to other people / teams via public Azure Blob or Docker Hub, it could be the legal issue.
I don't have much context here but it is how I understand the problem.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
